src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -54,122 +65,186 @@ const getMessageForTask = (isCurrentUser: boolean, taskType: TaskType) => {
+                        handleEditClick={this.handleEditClick}
+                        permissions={taskPermissions}
+                        tagged_message={description}
+                        translatedTaggedMessage={translatedTaggedMessage}
+                        translations={translations}
+                        type={COMMENT_TYPE_TASK}
+                        getAvatarUrl={getAvatarUrl}
+                        getUserProfileUrl={getUserProfileUrl}
+                        mentionSelectorContacts={mentionSelectorContacts}
+                        getMentionWithQuery={getMentionWithQuery}
+                        userHeadlineRenderer={userLinkInstance => {
+                            return (
+                                <FormattedMessage
+                                    {...getMessageForTask(!!currentUserAssignment, task_type)}
+                                    values={{ user: userLinkInstance }}
+                                />
+                            );
+                        }}
+                    />
+                    <div className="bcs-task-content bcs-task-status-container">
+                        {!!due_at && <TaskDueDate dueDate={due_at} status={status} />}
+                        <TaskStatus status={status} />
+                    </div>
+                    <div className="bcs-task-content">
+                        <AvatarGroup getAvatarUrl={getAvatarUrl} maxAvatars={3} users={assigned_to} />
+                    </div>
+                    <div className="bcs-task-content">
+                        {currentUserAssignment && shouldShowActions && (
+                            <TaskActions
+                                taskType={task_type}
+                                onTaskApproval={
+                                    isPending
+                                        ? noop
+                                        : () => onAssignmentUpdate(id, currentUserAssignment.id, TASK_NEW_APPROVED)
+                                }
+                                onTaskReject={
+                                    isPending
+                                        ? noop
+                                        : () => onAssignmentUpdate(id, currentUserAssignment.id, TASK_NEW_REJECTED)
+                                }
+                                onTaskComplete={
+                                    isPending
+                                        ? noop
+                                        : () => onAssignmentUpdate(id, currentUserAssignment.id, TASK_NEW_COMPLETED)
+                                }
                             />
-                        );
+                        )}
+                    </div>
+                </div>
+                <TaskModal
+                    editMode={TASK_EDIT_MODE_EDIT}
+                    error={modalError}
+                    feedbackUrl={getFeatureConfig(features, 'activityFeed.tasks').feedbackUrl || ''}
```

Comment:
Can the feature detection occur in the TaskModal itself? It looks like the `feedbackUrl` prop is required, anyway.

Reply:
I think so. existing code was using the feature check outside the modal, so initially didn't want to couple that refactoring with this PR. that would simplify the interface overall